Authenticator/Source/TokenManager.swift

```diff
@@ -96,53 +74,58 @@ class TokenManager {
         return Array(periods).sort()
     }
 
-    func addToken(token: Token) -> Bool {
-        guard let newKeychainItem = addTokenToKeychain(token) else {
-            return false
-        }
-        keychainItems.append(newKeychainItem)
+    func addToken(token: Token) throws {
+        let newPersistentToken = try keychain.addToken(token)
+        persistentTokens.append(newPersistentToken)
         saveTokenOrder()
-        return true
     }
 
-    func keychainItemAtIndex(index: Int) -> Token.KeychainItem {
-        return keychainItems[index]
+    func persistentTokenAtIndex(index: Int) -> PersistentToken {
+        return persistentTokens[index]
     }
 
-    func saveToken(token: Token, toKeychainItem keychainItem: Token.KeychainItem) -> Bool {
-        guard let newKeychainItem = updateKeychainItem(keychainItem, withToken: token) else {
-            return false
-        }
+    func saveToken(token: Token, toPersistentToken persistentToken: PersistentToken) throws {
+        let updatedPersistentToken = try keychain.updatePersistentToken(persistentToken, withToken: token)
```

Comment:
Length Violation: Line should be 100 characters or less: currently 106 characters